src/backend/cod1.c

```diff
@@ -1335,7 +1335,15 @@ code *getlvalue(code *pcs,elem *e,regm_t keepmsk)
             pcs->Irm = modregrm(3,0,s->Sreglsw & 7);
             if (s->Sreglsw & 8)
                 pcs->Irex |= REX_B;
-            if (e->EV.sp.Voffset == 1 && sz == 1)
+            if (e->EV.sp.Voffset == REGSIZE && sz == REGSIZE)
+            {
+                pcs->Irm = modregrm(3,0,s->Sregmsw & 7);
+                if (s->Sregmsw & 8)
+                    pcs->Irex |= REX_B;
+                else
+                    pcs->Irex &= ~REX_B;
```

Comment:
Ah sorry, now I spot the difference, Sregmsw instead of Sreglsw. All this undocumented hard to understand abbreviations, are a major hurdle to read backend code. Those were the low and high register for a fat struct, right.